From: =?utf-8?b?0KDRg9GB0LvQsNC9INCY0LbQsdGD0LvQsNGC0L7Qsg==?=
 <lrn1986@gmail.com>
Date: Thu, 16 Aug 2018 10:06:33 +0000
Subject: W32: tighten access rights requested to read symlink

According to
http://blog.kalmbach-software.de/2008/02/28/howto-correctly-read-reparse-data-in-vista/
we only need FILE_READ_EA, and should also use wider share flags.

Origin: upstream, 2.59.0, commit:256e74124821693c847b905dcbd70efae06c9809
---
 glib/gstdio.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/glib/gstdio.c b/glib/gstdio.c
index 6026696..1a56b19 100644
--- a/glib/gstdio.c
+++ b/glib/gstdio.c
@@ -466,8 +466,9 @@ _g_win32_readlink_utf16_raw (const gunichar2 *filename,
    * point and use DeviceIoControl() on it.
    */
   h = CreateFileW (filename,
-                   FILE_READ_ATTRIBUTES | SYNCHRONIZE | GENERIC_READ,
-                   FILE_SHARE_READ, NULL, OPEN_EXISTING,
+                   FILE_READ_EA,
+                   FILE_SHARE_READ|FILE_SHARE_WRITE|FILE_SHARE_DELETE,
+                   NULL, OPEN_EXISTING,
                    FILE_ATTRIBUTE_NORMAL
                    | FILE_FLAG_OPEN_REPARSE_POINT
                    | (attributes & FILE_ATTRIBUTE_DIRECTORY ? FILE_FLAG_BACKUP_SEMANTICS : 0),
